cpp/log/cluster_state_controller.h

```diff
@@ -0,0 +1,92 @@
+#ifndef CERT_TRANS_LOG_CLUSTER_STATE_CONTROLLER_H_
+#define CERT_TRANS_LOG_CLUSTER_STATE_CONTROLLER_H_
+
+#include <condition_variable>
+#include <functional>
+#include <map>
+#include <string>
+
+#include "log/etcd_consistent_store.h"
+#include "proto/ct.pb.h"
+#include "util/executor.h"
+#include "util/masterelection.h"
+
+namespace cert_trans {
+
+
+// A class which updates & maintains the states of the individual cluster
+// member nodes, and uses this information to determine the overall serving
+// state of the cluster.
+//
+// In particular, this class calculates the optimal STH for the cluster to
+// serve at any given time.
+template <class Logged>
+class ClusterStateController {
+ public:
+  ClusterStateController(util::Executor* executor,
+                         ConsistentStore<Logged>* store,
+                         const MasterElection& election,
+                         const int min_serving_nodes,
```

Comment:
As in, it'd be in a configuration key somewhere, so that everyone is in agreement on what these should be?